File: minebot/src/websocket/schemas/response/player_status_check.py
from pydantic import Field, model_validator

from ..base import BaseSchema, ResponseAwaitableSchema


class PlayerStatusCheckSchema(BaseSchema, ResponseAwaitableSchema, action="player-status-check"):
    """Schema for the player-status-check action."""

    username: str | None = Field(default=None, max_length=16)
    uuid: str | None = Field(default=None, max_length=36)
    online: bool | None = Field(default=None)

    @model_validator(mode="after")
    def validate_username_or_uuid(self) -> "PlayerStatusCheckSchema":
        if self.username is None and self.uuid is None:
            raise ValueError("Either 'username' or 'uuid' must be provided.")
        return self

File: minebridge/minebridge-core/src/main/java/org/mineacademy/minebridge/core/schema/PlayerStatusCheck.java
package org.mineacademy.minebridge.core.schema;

import org.mineacademy.minebridge.core.internal.BaseSchema;

import lombok.Getter;

public class PlayerStatusCheck extends BaseSchema {

    @Getter
    private final String username;

    @Getter
    private final String uuid;

    @Getter
    private final Boolean online;

    public PlayerStatusCheck(String username, String uuid, Boolean online) {
        super("player-status-check", false);
        this.username = username;
        this.uuid = uuid;
        this.online = online;
    }

}
